packages/blocks/src/api/registration.js

```diff
@@ -24,6 +24,101 @@ import { select, dispatch } from '@wordpress/data';
 import { isValidIcon, normalizeIconObject } from './utils';
 import { DEPRECATED_ENTRY_KEYS } from './constants';
 
+/**
+ * TODO: Editor block attributes object.
+ *
+ * @typedef {Object} WPBlockAttributeOptions
+ *
+ * @property {WPBlockAttributeType} type ...
+ * @property {string} source ...
+ * @property {string} selector ...
+ * @property {string} attribute ...
+ * @property {string} meta Attributes may be obtained from a post’s meta rather than from the block’s representation in saved post content. For this, an attribute is required to specify its corresponding meta key under the meta key
+ */
+
+/**
+ * Editor block category options.
+ *
+ * @typedef {Object.<string, WPBlockAttributeOptions>} WPBlockAttributes
+ */
+
+/**
+ * TODO: Editor block style object.
+ *
+ * @typedef {Object} WPBlockStyle
+ *
+ * @property {string}  name      Unique key name for the style definition.
+ * @property {string}  label     Label shown to the user in the editor.
+ * @property {boolean} isDefault Whether or not the style is selected by
+ *                               default.
+ */
+
+/**
+ * Editor block category options.
+ *
+ * @typedef {('common'|'formatting'|'layout'|'widgets'|'embed')} WPBlockCategories
```

Comment:
Do you have a reference for this syntax of defining options of a set that you can share? 

I can't find anything about it on the [`@type` documentation](https://jsdoc.app/tags-type.html), and my experience with similar options like [`@enum`](https://jsdoc.app/tags-enum.html) are they aren't quite what we're looking for for what you're expressing here.

Reply:
I had to dig for it, but this came from https://github.com/jsdoc/jsdoc/issues/629#issuecomment-58766768. So it's not documented as the "official" solution but it is parsed correctly, and this seemed more appropriate than an enum here.

Reply:
Makes sense to me!

The only thing I might suggest is that if a block is to be registered with a single category, it would make sense that this be singular "Category" as well, vs. "Categories"? Similar to `WPBlockAttributeType` and `WPBlockAttributeSource`.